### PR TITLE
[Bugfix] Dbt docs operator should not look for `graph.gpickle` file when `--no-write-json` is passed.

### LINKED ISSUE #882: [Bug] Dbt docs operator should not look for `graph.gpickle` file when `--no-write-json` is passed.
TLDR: `--no-write-json` prevents writing the `graph.gpickle` file. When this flag is there, that file should not be looked for.

Strictly speaking, `graph.gpickle` is not actually necessary for the dbt docs, so there is no harm to users who need to generate docs in respecting the `--no-write-json` flag.

### PR DESCRIPTION
**Pull request by dwreeves**
Fixes #882

### CODE REVIEW

**Review by tatiana — COMMENTED** [review]
Thanks for pointing out and fixing the issue, @dwreeves !  A minor question inline

File: cosmos/operators/local.py [hunk 1]
```diff
@@ -578,11 +578,13 @@ def __init__(self, **kwargs: Any) -> None:
...
 
     def check_static_flag(self) -> None:
-        flag = "--static"
         if self.dbt_cmd_flags:
-            if flag in self.dbt_cmd_flags:
+            if "--static" in self.dbt_cmd_flags:
                 # For the --static flag we only upload the generated static_index.html file
                 self.required_files = ["static_index.html"]
+        if self.dbt_cmd_global_flags:
+            if "--no-write-json" in self.dbt_cmd_global_flags and "graph.gpickle" in self.required_files:
```
[the remark this answers is not in the document]
Comment (dwreeves) [reply cosmos/operators/local.py:586]: Honestly, I do think it makes some sense to just remove it, if you don't mind the very minor backwards incompatibility. It's completely unnecessary for docs. This file is not really used for anything.

**Comment by dwreeves** [comment]
@tatiana 

- I removed `graph.gpickle` entirely, and opened #894 to discuss more generally how artifact uploads should be handled.
- I noticed in the tests that `test_operator_execute_deps_parameters` does not `assert` one of its conditions. However when I add the `assert`, it fails 🤔 This is a separate issue entirely from what I'm trying to do, but I found it curious and felt like I should note it.
image

File: cosmos/operators/local.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tatiana) [reply cosmos/operators/local.py:586]: Let's do this, then!
I noticed that the code coverage for this part of the code is a bit below the threshold of the project. Is there any chance you could improve this as part of the PR?
Once we have this, we can release a new 1.4 alpha with all the fixes.

File: cosmos/operators/local.py [hunk 2]
```diff
@@ -570,19 +570,21 @@ class DbtDocsLocalOperator(DbtLocalBaseOperator):
...
+                self.required_files.remove("graph.gpickle")
```
Comment (tatiana) [inline cosmos/operators/local.py:587]: I was about to merge, but just wanted to confirm: @dwreeves is this line still necessary?

Comment (tatiana) [reply cosmos/operators/local.py:587]: Thanks @dwreeves , let's keep it for now - time will tell if we should move out of this!

**Pull request merged**